be/src/runtime/fragment_mgr.cpp

```diff
@@ -997,25 +997,16 @@ void FragmentMgr::cancel_instance(const TUniqueId instance_id, const Status reas
     std::shared_ptr<PlanFragmentExecutor> non_pipeline_ctx;
     {
         std::lock_guard<std::mutex> state_lock(_lock);
-        const bool is_pipeline_instance = _pipeline_map.contains(instance_id);
-        if (is_pipeline_instance) {
-            auto itr = _pipeline_map.find(instance_id);
-            if (itr != _pipeline_map.end()) {
-                pipeline_ctx = itr->second;
-            } else {
-                LOG(WARNING) << "Could not find the pipeline instance id:" << print_id(instance_id)
-                             << " to cancel";
-                return;
-            }
+        CHECK(!_pipeline_map.contains(instance_id))
```

Comment:
DO NOT use check to avoid core online.
Using DCHECK and print warning message please.